R/interpolate.R

```diff
@@ -59,24 +59,26 @@ interpolate_file <- function(path, ..., .envir = parent.frame()) {
 #' @param package Package name.
 #' @rdname interpolate
 #' @export
-interpolate_package <- function(
-  package,
-  path,
-  ...,
-  .envir = parent.frame()
-) {
+interpolate_package <- function(package, path, ..., .envir = parent.frame()) {
+  check_string(package)
   check_string(path)
 
-  path_pkg <- system.file("prompts", path, package = package)
+  prompts_path <- file.path(inst_path(package), "prompts")
+  if (!dir.exists(prompts_path)) {
+    cli::cli_abort(
+      "{.pkg {package}} does not have a {.field prompts/} directory."
+    )
+  }
 
-  if (!nzchar(path_pkg)) {
+  prompt_path <- file.path(prompts_path, path)
+  if (!file.exists(prompt_path)) {
     cli::cli_abort(c(
       "{.pkg {package}} does not have {.val {path}} in its {.field prompts/} directory.",
       "i" = 'Run {.run dir(system.file("prompts", package = "{package}"))} to see available prompts.'
     ))
   }
 
-  interpolate_file(path_pkg, ..., .envir = .envir)
+  interpolate_file(prompt_path, ..., .envir = .envir)
 }
 
 read_file <- function(path) {
```

Comment:
The [Windows CI check failure](https://github.com/tidyverse/ellmer/actions/runs/18530627315/job/52812127827) reminded me that maybe you'd want `read_file()` to enforce UTF-8 encoding? These are the helpers we use in Shiny: https://github.com/wch/staticimports/blob/main/inst/staticexports/utf8.R